third_party/terraform/resources/resource_bigquery_table.go.erb

```diff
@@ -62,6 +63,154 @@ func resourceBigQueryTable() *schema.Resource {
 				Computed: true,
 			},
 
+			// ExternalDataConfiguration [Optional] Describes the data format,
+			// location, and other properties of a table stored outside of BigQuery.
+			// By defining these properties, the data source can then be queried as
+			// if it were a standard BigQuery table.
+			"external_data_configuration": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Autodetect : [Required] Let BigQuery try to autodetect the
+						// schema and format of the table.
+						"autodetect": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						// BigtableOptions: [Optional] Additional options if sourceFormat is set to BIGTABLE.
+                                                "bigtable_options": {
```

Comment:
I think this is a `TypeList` or `TypeSet` in the API?

Reply:
> externalDataConfiguration.bigtableOptions | nested object

https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#resource

But since I implement this as an inline JSON (similar to `schema`) I think it should be `TypeString`?

Reply:
I implemented all options for Bigtable except for `column_families` which is a JSON string, similar to `schema`.